tmail-backend/integration-tests/imap/memory-imap-integration-tests/src/test/java/com/linagora/tmail/imap/IMAPTeamMailboxIntegrationTest.java

```diff
@@ -412,4 +413,37 @@ void topFolderCreatedByBobShouldBeAccessibleByAlice() throws Exception {
             .sendCommand("LIST \"\" \"*\""))
             .contains("* LIST (\\HasNoChildren) \".\" \"#TeamMailbox.marketing.new1\"");
     }
+
+    @Test
+    void topFolderCreatedByMemberAShouldBeAccessibleByOtherUserWhenHasRight(GuiceJamesServer server) throws Exception {
+        // Verify user `other3` is not a member of the team mailbox `marketing`
+        assertThat(CollectionConverters.asJava(server.getProbe(TeamMailboxProbe.class)
+            .listMembers(MARKETING_TEAM_MAILBOX)))
+            .doesNotContain(OTHER3)
+            .contains(MINISTER);
+
+        // Verify that the team mailbox `marketing.new1` does not list in the mailbox list of user `other3`
+        assertThat(testIMAPClient.connect(IMAP_HOST, imapPort)
+            .login(OTHER3, OTHER3_PASSWORD)
+            .sendCommand("LIST \"\" \"*\""))
+            .doesNotContain("\"#TeamMailbox.marketing.new1\"");
+
+        // Given `minister` creates the team mailbox `marketing.new1`
+        assertThat(testIMAPClient.connect(IMAP_HOST, imapPort)
+            .login(MINISTER, MINISTER_PASSWORD)
+            .sendCommand("CREATE #TeamMailbox.marketing.new1"))
+            .contains("CREATE completed");
+
+        // When add the right `other3` to the team mailbox `marketing.new1`
+        server.getProbe(ACLProbeImpl.class)
+            .replaceRights(MARKETING_TEAM_MAILBOX.mailboxPath("new1"),
+                OTHER3.asString(),
+                MailboxACL.Rfc4314Rights.of(List.of(MailboxACL.Right.Lookup, MailboxACL.Right.Read)));
+
+        // Then the team mailbox `marketing.new1` should list in the mailbox list of user `other3`
+        assertThat(testIMAPClient.connect(IMAP_HOST, imapPort)
+            .login(OTHER3, OTHER3_PASSWORD)
+            .sendCommand("LIST \"\" \"*\""))
+            .contains("* LIST (\\HasNoChildren) \".\" \"#TeamMailbox.marketing.new1\"");
+    }
```

Comment:
Same tests when bob creates `a.b.c.d` please let's verify rights on `a`, `a.b`, `a.b.c` and `a.b.c.d`